application/src/main/java/org/togetherjava/tjbot/commands/system/CommandProvider.java

```diff
@@ -0,0 +1,61 @@
+package org.togetherjava.tjbot.commands.system;
+
+import org.togetherjava.tjbot.commands.BotCommand;
+import org.togetherjava.tjbot.commands.UserInteractor;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Provides all registered commands.
+ */
+public interface CommandProvider {
+    /**
+     * Gets a list of all currently available and registered interactors.
+     *
+     * @return all interactors
+     */
+    Collection<UserInteractor> getInteractors();
+
+    /**
+     * Gets a list of all currently available and registered interactors.
+     *
+     * @return all interactors
+     */
+    default Collection<BotCommand> getCommands() {
+        return getInteractors().stream()
+            .filter(BotCommand.class::isInstance)
+            .map(BotCommand.class::cast)
+            .toList();
+    }
+
+    /**
+     * Gets the slash command registered under the given name, if any.
+     *
+     * @param name the name of the command (including its prefix)
+     * @return the command registered under this name, if any
+     */
+    Optional<UserInteractor> getInteractor(String name);
+
+    /**
+     * Gets the slash command registered under the given name, if any.
+     *
+     * @param name the name of the command
+     * @param type the type of the command
+     * @return the command registered under this name, if any
+     */
+    <T extends UserInteractor> Optional<T> getInteractor(String name,
+            @Nullable Class<? extends T> type);
+
+    /**
+     * Gets any interactors with the given name. The name you give should be without prefix, these
+     * methods checks all prefixes themselves. If your name is with prefix, use
+     * {@link #getInteractor(String)}
+     *
+     * @param name the name of the command
+     * @return a list of commands with that name
+     */
+    List<UserInteractor> getAnyInteractors(final String name);
+}
```

Comment:
The UX here is confusing and easy to get wrong. Maybe commands, to the user, shouldnt be addressable by string but by `record InteractorName(String prefix, String commandName){}` and then `getInteractor` should take `InteractorName`, while this one takes `String commandName` or similar.

Reply:
i would still want to rename the param to `commandName` to emphasis that its not the full name (with prefix). its just a bit hard to understand the difference between all these "names" when everything is just `String`.

that said, something about the method name feels weird to me. i dont think `Any` is the idiomatic wording to use, maybe `All`? It gives a list of all interactors with the given "base name" after all. `Any` sounds as if it picks a single of them, similar to `stream().filter(...).findAny()`

Reply:
Prefixed arguments have been renamed to prefixedName

Non prefixed arguments stay as "name"